parquet-column/src/main/java/org/apache/parquet/filter2/recordlevel/IncrementallyUpdatedFilterPredicate.java

```diff
@@ -145,6 +146,83 @@ public boolean accept(Visitor visitor) {
     }
   }
 
+  /**
+   * A ValueInspector implementation that keeps state for one or more delegate inspectors.
+   */
+  abstract static class DelegatingValueInspector extends ValueInspector {
```

Comment:
I created this because I thought the generated `IncrementallyUpdatedFilterPredicateBuilder` was getting complex in terms of LOC -- all the `Contains` ValueInspectors were overriding `public void update(int|float|binary|double|... value)` as well as `reset()`, and it was getting hard to reason about. However.... this is probably a bit less efficient, since the delegate ValueInspectors are accessed via iterator loops rather than directly invoking `left`/`right`. Let me know what you think, I'm happy to change it back to the way it was

Reply:
`IncrementallyUpdatedFilterPredicateGenerator.java` is pretty difficult to read and review. Since this is only applied to `ContainsPredicate` and does not affect others, I'm fine to make this change.